contents/docs/tutorials/spa.md

```diff
@@ -0,0 +1,268 @@
+
+We will be using this function to periodically check if certain elements are visible as the user goes through the page.
+
+**Determining elements that represent new sections**
+
+The next step is to decide what elements in our SPA constitute a new section/screen. We will use these to trigger pageviews when they are visible for the user.
+
+For example, these could be section heders, as well as wrapping `div` tags. This will vary from website to website, and you need to decide for yourself when _exactly_ the page view should be triggered. For example, do you want to trigger a `$pageview` when the section is partly visible or entirely covering the screen? 
+
+Once you have determined the elements to use, you can specify them in a structure like this:
+
+```js
+let elementsToTrack = [
+    {
+        name: "section1",
+        ref: document.getElementById('section1'),
+        visible: true
+    },
+    {
+        name: "section2",
+        ref: document.getElementById('section2'),
+        visible: false
+    },
+]
+```
+
+This structure will be useful for the next step, but you can of course customize this entire solution as you wish.
+
+**Listening for changes to the window to trigger page views**
+
+With the structure in place, let's now actually listen for changes and capture events.
+
+First, let's write up a handler that checks if our elements are visible and captures a pageview if they are:
+
+```js
+const visibilityChangeHandler = () => {
+    
+    // Loop through the relevant elements
+    for (let el of elementsToTrack) {
+
+        // Check if the element is visible
+        let visible = isElementInViewport(el.ref)
+
+        // Check diff to prevent useless triggers
+        if (visible !== el.visible) {
+
+            // Update the element visibility state
+            el.visible = visible
+
+            // Capture a pageview with PostHog
+            if (visible) {
+                posthog.capture(
+                    '$pageview', 
+                    { section: el.name, $current_url: el.name }
+                )
+            }
+        }
+    }
+}
+```
+
+The comments in the code provide context as to what each line is doing, but, essentially, when the handler is triggered, we check the elements we care about for _changes_ in visibility, update the state, and capture a `$pageview` if the change was from `false -> true` only. 
+
+You could also add some more complex logic here, such as capturing a pageview when one element stops being visible and another one becomes visible. 
+
+Additionally, you also have choices when calling `posthog.capture`. You can either set your custom property to track the current screen (`section` in the example), or override PostHog's default `$current_url` prop (or both). 
```

Comment:
I'd probably make a recommendation here to be more precise, or at least explain the pros/cons of each.

Specifically, it makes sense to me to use current_url so paths works for a SPA

Reply:
There is a recommendation a few lines down, but I can maybe make that more clear (+ mention paths)